hpat/datatypes/hpat_pandas_series_functions.py

```diff
@@ -230,6 +230,37 @@ def hpat_pandas_series_index_impl(self):
     return hpat_pandas_series_index_impl
 
 
+@overload_attribute(SeriesType, 'size')
+def hpat_pandas_series_size(self):
+    """
+    Pandas Series attribute :attr:`pandas.Series.size` implementation
+
+    **Algorithm**: result = series.size
+
+    **Test**: python -m hpat.runtests hpat.tests.test_series.TestSeries.test_series_size
+
+    Parameters
+    ----------
+    series: :obj:`pandas.Series`
+        input series
+
+    Returns
+    -------
+    :class:`pandas.Series`
+        Return the number of elements in the underlying data.
+    """
+
+    _func_name = 'Attribute size.'
+
+    if not isinstance(self, SeriesType):
+        raise TypingError('{} The object must be a pandas.series. Given: {}'.format(_func_name, self))
+
+    def hpat_pandas_series_size_impl(self):
+        return self._data.size
```

Comment:
I would propose to use `len(self._data)` instead.

Reply:
As I can see playing with Pandas in console: pd.Series() does not store _data = None in internal representation. So _data is always something even if Series is empty.
Also this could be replaced with `len(self._data)`.